internal/uploader/events_uploader_handler.go

```diff
@@ -0,0 +1,247 @@
+package uploader
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"net/textproto"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/openshift/assisted-service/internal/common"
+	eventsapi "github.com/openshift/assisted-service/internal/events/api"
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/pkg/k8sclient"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+var _ UploadHandler = &EventsUploader{}
+
+type EventsUploader struct {
```

Comment:
Maybe the file name should match?

`events_uploader.go`?

Reply:
Also, does this need to be exported since we have an interface?

Reply:
True, modified it to reflect this!